docs/_ruleset_functions.md

```diff
@@ -1,33 +1,270 @@
+
+**Returns** ```return surface_conditioning_category_dict```  
 
 ## get_opaque_surface_type
-Description:  This function would cycle through each surface and determine whether it is a wall, ceiling or floor.  
+Description: This function would determine whether it is a wall, ceiling or floor.  
+
+Inputs:
+  - **Surface**: The surface that needs to determine surface type.  
+
+Returns:
+- **opaque_surface_type**: The Opaque Surface Type [roof, heated slab-on-grade, unheated slab-on-grade, floor, above-grade wall, below-grade wall].  
+
+
+Logic:  
+
+- If surface tilt is more than or equal to 0 degree and less than 60 degrees, surface is classified as roof: ```if 0 <= surface.tilt < 60: surface_type = "ROOF"```  
+
+- Else if surface tile is more than 120 degrees and less than or equal to 180 degrees, surface is classified as floor: ```else if 120 < surface.tilt <= 180:```  
+
+  - Determine if surface is heated slab-on-grade: ```if ( surface.construction.has_radiant_heating ) AND ( surface.adjacent_to == "GROUND" ): surface_type = "HEATED SLAB-ON-GRADE"```  
+
+  - Else determine if surface is unheated slab-on-grade: ```else if surface.adjacent_to == "GROUND": surface_type = "UNHEATED SLAB-ON-GRADE"```  
+
+  - Else, surface is floor: ```else: surface_type = "FLOOR"```  
+
+- Else, surface is classified as wall: ```else:```  
+
+  - Determine if surface is below grade wall: ```if surface.adjacent_to == "GROUND": surface_type = "BELOW-GRADE WALL"```  
+
+  - Else determine if surface is above grade wall: ```else: surface_type = "ABOVE-GRADE WALL"```  
+
+**Returns** ```return surface_type```  
+
+## get_wwr
+
+Description: This function would determine window wall ratio for a building segment.  
+
+Inputs:
+  - **RMR**: The RMR that needs to determine window wall ratio.  
+
+Returns:
+- **window_wall_ratio**: The window wall ratio of the building segment.  
+
+Function Call:
+
+- get_zone_conditioning_category()
+- get_surface_conditioning_category()
+- get_opaque_surface_type()
+
+Logic:
+
+- Get zone conditioning category dictionary for RMR: `zcc_dictionary = get_zone_conditioning_category(RMR)`
+
+- Get surface conditioning category dictionary for RMR: `scc_dictionary = get_surface_conditioning_category(RMR)`
+
+- For each building segment in the model: `for building_segment in RMR.building.building_segments:`
+
+  - For each thermal_block in building segment: `for thermal_block in building_segment.thermal_blocks:`
+
+    - For each zone in thermal block: `zone in thermal_block.zones:`
+
+      - Check if zone is conditioned or semi-heated: `if zcc_dictionary[zone.id] in [CONDITIONED RESIDENTIAL, CONDITIONED NON-RESIDENTIAL, CONDITIONED MIXED, SEMI-HEATED]`
+
+        - For each surface in zone: `for surface in zone.surfaces:`
+
+          - If surface is above-grade wall and adjacent to the exterior: `if ((get_opaque_surface_type(surface) == "ABOVE-GRADE WALL") AND (surface.adjacent_to == "EXTERIOR")):`  
+
```

Comment:
We have pending interpretation request about this. Based on discussions of the ECB subcommittee, it seems likely that we need to drop "adjacent_to=exterior condition". 
So for this function we'll need to get_surface_conditioning_category, and for all surfaces that are exterior res, exterior non res, exterior mixed and semi-exterior, and include all surfaces that have get_opaque_surface_type=wall.

Reply:
I was referencing G3.1-5(c) vertical fenestration...separates conditioned or semi-heated space from exterior. But in the current get_surface_conditioning_category, if a surface is semi-exterior, it could be between conditioned and unconditioned spaces, which is not exterior. And if a surface is exterior res or exterior non res, it could be between conditioned and unenclosed spaces.

Reply:
I submitted interpretation request to ASHRAE about this, and based on ECB subcommittee discussions it is all but certain that the interpretation will say that both exterior and semi-exterior surfaces in your example above count as adjacent to exterior because the unconditioned or unenclosed spaces are themselves pathway to exterior.

Reply:
@mkarpman Its been a while since we've discussed this. We have two WWR calculations, one for determining baseline U/SHGC- so this needs to be a single value for the entire building and one for determining the actual baseline WWR (this needs to be by building segment). Here I see we're doing the WWR calculation by building segment, so this is for purpose #2  correct?

Reply:
Correct. The interpretation request that I submitted asks for clarification on both WWR calculations. #2 above has been discussed by the subcommittee, but we haven't talked about #1 yet.

Reply:
Surface.adjacent_to is now replaced with get_surface_conditioning_category outputs. @mkarpman